Mark libwebrtc as a product dependency of WebCore

youennf opened the pull request:
#### 0ed3e7ebcd2a6226206ed1a4cc4ee62a603c67d1
<pre>
Mark libwebrtc as a product dependency of WebCore
<a href="https://rdar.apple.com/128785444">rdar://128785444</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=274747">https://bugs.webkit.org/show_bug.cgi?id=274747</a>

Reviewed by Elliott Williams.

XCode does not always know that libwebrtc.dylib is a pre-requisite to build WebCore.
We add it as a product dependency of the generate unified files step similarly to what has been done for libWTF.a.
We restrict this to iOS, macOS and tvOS. This should be a no-op for production builds.

* Source/WebCore/Configurations/Base.xcconfig:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/279635@main">https://commits.webkit.org/279635@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1de5c1482f948d2ad00d50c89659f63bb34eecfb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53326 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32677 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5825 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56605 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4052 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55632 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40084 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3805 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43234 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2652 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55424 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/30860 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46059 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24364 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/27753 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3379 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2208 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/49505 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3547 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58201 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28475 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3541 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50633 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29690 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46267 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49969 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11776 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30614 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29453 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (changes requested):
> XCode does not always know that libwebrtc.dylib is a pre-requisite to build WebCore

It should! Implicit dependency resolution should detect the dependency via the linker flag we pass here:
https://github.com/WebKit/WebKit/blob/33808df7d0b1b590b321d6c6321386652f029cda/Source/WebCore/Configurations/WebCore.xcconfig#L134


And in a recent build <https://build.webkit.org/#/builders/938/builds/8822>, I see that it is indeed being picked up. Near the top of the build log:
```
    Target 'WebCore' in project 'WebCore'
        ➜ Explicit dependency on target 'PAL' in project 'PAL'
        ➜ Explicit dependency on target 'Derived Sources' in project 'WebCore'
        ➜ Explicit dependency on target 'Generate Unified Sources' in project 'WebCore'
        ➜ Implicit dependency on target 'JavaScriptCore' in project 'JavaScriptCore' via file 'JavaScriptCore.framework' in build phase 'Link Binary'
        ➜ Implicit dependency on target 'ANGLE (dynamic)' in project 'ANGLE' via options '-weak-lANGLE-shared' in build setting 'OTHER_LDFLAGS'
        ➜ Implicit dependency on target 'WebGPU' in project 'WebGPU' via options '-weak_framework WebGPU' in build setting 'OTHER_LDFLAGS'
        ➜ Implicit dependency on target 'libwebrtc' in project 'libwebrtc' via options '-weak-lwebrtc' in build setting 'OTHER_LDFLAGS'
```

So, I don't think this is necessary. If we need webrtc to be built before running WebCore's "Generate Unified Sources" script, then I understand how this would affect the build, but I don't understand why we would need to do that. Is there something I'm missing?

emw-apple reviewed (approved):
Youenn gave me an example of where this is needed internally. Unfortunately, we don't have a way to add `-l` options to OTHER_LDFLAGS using WebKitAdditions, so we have to make this dependency on webrtc explicit, even in situations where we won't link it.

youennf committed:
Mark libwebrtc as a product dependency of WebCore
rdar://128785444
https://bugs.webkit.org/show_bug.cgi?id=274747

Reviewed by Elliott Williams.

XCode does not always know that libwebrtc.dylib is a pre-requisite to build WebCore.
We add it as a product dependency of the generate unified files step similarly to what has been done for libWTF.a.
We restrict this to iOS, macOS and tvOS. This should be a no-op for production builds.

* Source/WebCore/Configurations/Base.xcconfig:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/279635@main
